include/psa/crypto_extra.h

```diff
@@ -2106,6 +2106,8 @@ struct psa_pake_operation_s {
     unsigned int MBEDTLS_PRIVATE(id);
     /* Algorithm of the PAKE operation */
     psa_algorithm_t MBEDTLS_PRIVATE(alg);
+    /* A primitive of type compatible with algorithm */
+    psa_pake_primitive_t MBEDTLS_PRIVATE(primitive);
```

Comment:
One thing that I would suggest to consider is moving the new structure field to the end to conserve backward ABI compatibility.

Reply:
I'm not sure about moving this field to the end of the structure as locating it with `id` and `alg` values makes more sense. Then we have unions.

Reply:
Maybe it's a question to one of our gatekeepers? Whether we prefer backwards ABI compatibility or better readability.

Reply:
We end up breaking ABI compatibility in pretty much every release these days. So one more ABI break doesn't matter. (Except in LTS, there we want to preserve the ABI as well as the API.)

Beyond readability for grouping, there are other considerations for the ordering of struct fields. I haven't checked what might apply here.

* Avoid holes due to alignment. E.g. prefer `{char; char; short;}` or `{short; char; char}` to `{char; short; char;}` since the first two typically fill 4 bytes but the last one typically needs 6 bytes, with two padding bytes.
* For long structures: on Cortex-M, it takes less code to access the first 128 elements, so it's better to put often-accessed byte fields within the first 128 bytes, 2-byte fields within 256 bytes, 4-byte fields within 512 bytes. Here, “often-accessed” is about the number of places in the code that access the field, not about how frequent the accesses are at runtime. Of course the very first field is even cheaper. See https://github.com/Mbed-TLS/mbedtls/pull/5234
* For long structures: if some fields are often accessed at the same time, it helps if they're on the same cache line. The size of a cache line depends on the architecture, of course, but what this means is that it helps if they're close together.
* If some parts of a structure depend on compilation option, it's usually easier to follow things in a debugger if all the common parts are at the beginning.